Mathlib/NumberTheory/LSeries/Basic.lean

```diff
@@ -160,73 +236,103 @@ lemma LSeriesSummable_of_le_const_mul_rpow {f : ArithmeticFunction ℂ} {x : ℝ
     exact mem_image.mpr ⟨n, mem_range.mpr hn, rfl⟩
 
-theorem LSeriesSummable_of_bounded_of_one_lt_re {f : ArithmeticFunction ℂ} {m : ℝ}
-    (h : ∀ n : ℕ, Complex.abs (f n) ≤ m) {z : ℂ} (hz : 1 < z.re) : f.LSeriesSummable z := by
-  refine LSeriesSummable_of_le_const_mul_rpow hz ⟨m, fun n ↦ ?_⟩
-  simp only [norm_eq_abs, sub_self, Real.rpow_zero, mul_one, h n]
-#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_re ArithmeticFunction.LSeriesSummable_of_bounded_of_one_lt_re
-
-theorem LSeriesSummable_of_bounded_of_one_lt_real {f : ArithmeticFunction ℂ} {m : ℝ}
-    (h : ∀ n : ℕ, Complex.abs (f n) ≤ m) {z : ℝ} (hz : 1 < z) : f.LSeriesSummable z :=
-  LSeriesSummable_of_bounded_of_one_lt_re h <| by simp only [ofReal_re, hz]
-#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_real ArithmeticFunction.LSeriesSummable_of_bounded_of_one_lt_real
-
-open scoped ArithmeticFunction
-
-theorem zeta_LSeriesSummable_iff_one_lt_re {z : ℂ} : LSeriesSummable ζ z ↔ 1 < z.re := by
-  rw [← LSeriesSummable_iff_of_re_eq_re (Complex.ofReal_re z.re), LSeriesSummable, ←
-    summable_norm_iff, ← Real.summable_one_div_nat_rpow, iff_iff_eq]
-  by_cases h0 : z.re = 0
-  · rw [h0, ← summable_nat_add_iff 1]
-    apply congr rfl
-    ext n
-    simp [n.succ_ne_zero]
-  · apply congr rfl
-    ext ⟨- | n⟩
-    · simp [h0]
-    simp only [cast_zero, natCoe_apply, zeta_apply, succ_ne_zero, if_false, cast_succ, one_div,
-      Complex.norm_eq_abs, map_inv₀, Complex.abs_cpow_real, inv_inj, zero_add]
-    rw [← cast_one, ← cast_add, Complex.abs_natCast, cast_add, cast_one]
-#align nat.arithmetic_function.zeta_l_series_summable_iff_one_lt_re ArithmeticFunction.zeta_LSeriesSummable_iff_one_lt_re
+/-- If `f` is bounded, then its `LSeries` is summable at `s` when `re s > 1`. -/
+theorem LSeriesSummable_of_bounded_of_one_lt_re {f : ℕ → ℂ} {m : ℝ}
+    (h : ∀ n ≠ 0, Complex.abs (f n) ≤ m) {s : ℂ} (hs : 1 < s.re) :
+    LSeriesSummable f s := by
+  refine LSeriesSummable_of_le_const_mul_rpow hs ⟨m, fun n hn ↦ ?_⟩
+  simp only [norm_eq_abs, sub_self, Real.rpow_zero, mul_one, h n hn]
+#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_re LSeriesSummable_of_bounded_of_one_lt_re
+
+/-- If `f` is bounded, then its `LSeries` is summable at `s : ℝ` when `s > 1`. -/
+theorem LSeriesSummable_of_bounded_of_one_lt_real {f : ℕ → ℂ} {m : ℝ}
+    (h : ∀ n ≠ 0, Complex.abs (f n) ≤ m) {s : ℝ} (hs : 1 < s) :
+    LSeriesSummable f s :=
+  LSeriesSummable_of_bounded_of_one_lt_re h <| by simp only [ofReal_re, hs]
+#align nat.arithmetic_function.l_series_summable_of_bounded_of_one_lt_real LSeriesSummable_of_bounded_of_one_lt_real
+
+-- TODO: Move this to a separate file on concrete L-series
+
+open Set in
+/-- The `LSeries` with all coefficients `1` converges at `s` if and only if `re s > 1`. -/
+theorem LSeriesSummable.one_iff_one_lt_re {s : ℂ} : LSeriesSummable 1 s ↔ 1 < s.re := by
```

Comment:
Slight generalizations could be useful: if the coefficients do not go to 0, `re s > 1` is necessary ; 
if they are bounded, `re s > 1` is sufficient.

Reply:
If you take the indicator function of the subset of `k`th powers as `f`, then the L-series converges for `re s > 1/k`, but the coefficients do not go to zero. The second statement is there (`LSeriesSummable_of_bounded_of_one_lt_re`).

Reply:
you're perfectly right, the first (correct) lemma would just say that `re s > 0` is necessary, and it's not enough.